pkgs/development/python-modules/lintrunner/default.nix

```
@@ -0,0 +1,55 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  maturin,
+  rustPlatform,
+  cargo,
+  rustc,
+  python,
+}:
+
+buildPythonPackage rec {
+  pname = "lintrunner";
+  version = "0.12.7";
+  format = "pyproject";
+
+  # NOTE: there is a PR open on the upstream repository for the Cargo.lock file.
+  # After that is merged we can change the owner and rev to the upstream
+  # repository instead of a fork.
+  src = fetchFromGitHub {
+    owner = "PrestonHager";
+    repo = pname;
+    rev = "main";
+    #rev = "v${version}";
```

```suggestion
    tag = "v${version}";
```
and please use `fetchpatch2`.

---

Unsure what you mean by `fetchpatch2`? I'd like to use patches instead of a fork of the repository, but I could not get a `git diff --cached > file.patch` to work locally, probably doing something wrong.